### PR TITLE
Add missing profiler docs

### PR DESCRIPTION
**Pull request by davmason**
## Summary

Add profiler APIs from 5.0 to the docs

### CODE REVIEW

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipedefineevent-method.md
Comment (tommcdon) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipedefineevent-method.md:81]: Suggest adding a link to https://docs.microsoft.com/dotnet/core/diagnostics/eventpipe and https://docs.microsoft.com/dotnet/core/diagnostics/well-known-event-providers

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipecreateprovider-method.md
Comment (tommcdon) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipecreateprovider-method.md:43]: Suggest adding a link to https://docs.microsoft.com/dotnet/core/diagnostics/eventpipe and https://docs.microsoft.com/dotnet/core/diagnostics/well-known-event-providers

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipeaddprovidertosession-method.md
Comment (tommcdon) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipeaddprovidertosession-method.md:44]: Suggest adding a link to https://docs.microsoft.com/dotnet/core/diagnostics/eventpipe

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipegetproviderinfo-method.md
Comment (tommcdon) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipegetproviderinfo-method.md:52]: Suggest adding a link to https://docs.microsoft.com/dotnet/core/diagnostics/eventpipe and https://docs.microsoft.com/dotnet/core/diagnostics/well-known-event-providers

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipestartsession-method.md
Comment (tommcdon) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipestartsession-method.md:52]: same as above

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo12-interface.md
Comment (tommcdon) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo12-interface.md:38]: Suggest adding a link to https://docs.microsoft.com/dotnet/core/diagnostics/eventpipe and https://docs.microsoft.com/dotnet/core/diagnostics/well-known-event-providers

**Review by tommcdon — APPROVED** [review]
Left a few comments, otherwise LGTM

File: includes/net-core-50-md.md
Comment (sywhang) [inline includes/net-core-50-md.md:1]:
```suggestion
Available since .NET 5.0
```

**Review by sywhang — APPROVED** [review]
Thanks @davmason! I can merge this after you address the feedback here :-)

**Comment by davmason** [comment]
Don't merge for a bit, I just discovered the we use wildy different formatting in some areas of the profiler docs and I want to fix it

**Comment by sywhang** [comment]
> Don't merge for a bit, I just discovered the we use wildy different formatting in some areas of the profiler docs and I want to fix it

Sure, just let me know when it's ready for review then

**Comment by davmason** [comment]
@sywhang can you take a look now? We had 3-4 different ways we were formatting parameters, I picked what seemed to be the most common one and made them all consistent

**Review by sywhang — APPROVED** [review]
:shipit:

**Pull request merged**
